R/pkg/R/DataFrame.R

```diff
@@ -3905,6 +3905,16 @@ setMethod("rollup",
             groupedData(sgd)
           })
 
+isTypeAllowedForSqlHint <- function(x) {
+  if (is.character(x) || is.numeric(x)) {
+    TRUE
+  } else if (is.list(x)) {
+    all(sapply(x, (function(y) is.character(y) || is.numeric(y))))
```

Comment:
I look into this more deeply, I think this style seems a bit odd, as a nit, I think this should be

`all(sapply(x, function(y) { is.character(y) || is.numeric(y) } ))` 

think it's more readable this way. also see L2458 for an example https://github.com/apache/spark/blob/aec391c9dcb6362874736e663d435f9dd8400125/R/pkg/R/DataFrame.R#L2458

Reply:
also, if it is a `list` could we clarify if it is supposed to work with multiple hint in different types in that list (this might be "unique" to R), for example

```
> x <- list("a", 3)
> all(sapply(x, function(y) { is.character(y) || is.numeric(y) } ))
[1] TRUE

> x <- list("a", NA)
> all(sapply(x, function(y) { is.character(y) || is.numeric(y) } ))
[1] FALSE
```

Reply:
I think multiple hint in different types in a single list works OK. I tested 
```df.hint("hint1", Seq(1, 2, "a"))``` in scala, it works OK.